### ISSUE TITLE
What ABRiS could do that it couldn't do databricks at that moment?

### ISSUE THREAD

**Issue by xavtx1** [AbsaOSS/ABRiS#77]
Hello, I saw your ABRiS conference on the Databricks channel.

https://databricks.com/session/abris-avro-bridge-for-apache-spark

At the conference you explained what ABRiS could do that it couldn't do Databricks at that moment.

It's been more than a year since then.

Please could you write, which gap is still offering ABRiS to current date (mainly as far as confluent is concerned)

It would be very useful, as I believe that both ABRiS and Databricks have evolved since then.

Thank you

**Comment by felipemmelo** [comment]
Hi @xavtx1 , thanks for the question. I'll put a history line here.

Back in Jan 2018 we had the need to integrate Kafka with Spark through Avro + Schema Registry, but the only thing available out there was [this](https://github.com/databricks/spark-avro/blob/branch-4.0/README-for-old-spark-versions.md), which did not support anything closer to that. Than we developed and open-sourced ABRiS. 

A few months after the presentation you mentioned, Databricks delivered support for Avro also in the Apache version as described [here](https://spark.apache.org/docs/latest/sql-data-sources-avro.html). It did not, however, support Confluent or Schema Registry. 

Later on, Databricks deployed a version that supports Schema Registry as described [here](https://databricks.com/blog/2019/02/15/how-to-work-with-avro-kafka-and-schema-registry-in-databricks.html). However, it still does not support neither Confluent nor Schema Registry naming strategies as part of the API. The available naming strategies are [here](https://docs.confluent.io/current/schema-registry/serializer-formatter.html#group-by-topic-or-other-relationships).

Finally, to make things easier for everyone, we changed ABRiS API to make it compliant with Spark Avro's one so that users can seamlessly shift between them. This version is ABRiS 3.

To your question now, in the current state of affairs, you'd use ABRiS if you:

1. Need to connect to Confluent (id of schema on top of the payload)
2. Use naming strategies other than Topic Name.
3. Are not a customer of Databricks but needs Schema Registry, as to the best of my knowledge, only the paid version supports it.
